### PR TITLE
Fix trees MSE deprecation warning [1542]

### PR DESCRIPTION
**Pull request by willsmithorg**
*Issue #, if available:*
#1542 

*Description of changes:*
Rename 'mse' to 'squared_error' in parameters to sklearn::xt and sklearn::rf

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ Be sure to select the *Source* option from the installation preferences.
 
 - We recommend developing on Linux as this is the only OS where all features are currently 100% functional. Avoid introducing changes that will only work on a particular OS, as we're currently working to support MacOSX and Windows. Changes to existing code that improve cross-platform compatibility are most welcome!
 
-- Use Python >= 3.6 for development, as these are the only versions where AutoGluon is fully functional.
+- Use Python >= 3.7 for development, as these are the only versions where AutoGluon is fully functional.
```
Comment (Innixma) [inline CONTRIBUTING.md:52]: Can be more specific: Use Python 3.7, 3.8, and 3.9 for development,

**Review by Innixma — APPROVED** [review]
LGTM! Just had one minor comment. Thanks for the contribution!

File: CONTRIBUTING.md (hunk 1 above)
Comment (willsmithorg) [reply CONTRIBUTING.md]: Done.

Comment (willsmithorg) [reply CONTRIBUTING.md]: Unit test has failed on OOM.  Pretty sure unrelated to my code.

**Pull request merged**